tests/integration/test_mysql_reader.py

```diff
@@ -0,0 +1,34 @@
+import os
+from nck.readers.mysql_reader import MySQLReader
+
+
+class TestMySQlReader:
+
+    # using tests/resources/employees.sql database
+
+    basic_query = "SELECT * FROM firstname"
+
+    # +---------+
+    # | myfield |
+    # +---------+
+    # | Pierre  |
+    # | Julien  |
+    # +---------+
+
+    sql_env = {
+        "user" : os.environ["MYSQL_USER"],
+        "password" : os.environ["MYSQL_PASSWORD"],
+        "host" : os.environ["MYSQL_HOST"],
+        "port": os.environ["MYSQL_PORT"],
+        "database" : os.environ["MYSQL_DATABASE"]
+    }
+
+    sql_env["query"] = basic_query
+
+    mySQLReader = MySQLReader(**sql_env)
+
+    def test_one(self):
+        # MySQLReader.read() returns a generator object we need to extract
+        json_stream = list(self.mySQLReader.read())
+        request_result = list(json_stream[0].readlines())
```

Comment:
Can you test in the read method of the SQLReader to change
yield self._run_query()
into
return self._run_query()

In order to see if the "json_stream[0].readlines()" could be simplified into "json_stream.readlines()"

Reply:
I disagree, the whole philosophy of NCK is to work with generators, it should be kept that way to keep the software consistent

Reply:
Indeed, but we can still use :
yield from self._run_query()
instead of
yield self._run_query()